src/Microsoft.SqlTools.ServiceLayer/Profiler/ProfilerSessionMonitor.cs

```diff
@@ -186,113 +183,105 @@ private bool RemoveSession(int sessionId, out ProfilerSession session)
             }
         }
 
-        public void PollSession(int sessionId)
-        {
-            lock (this.sessionsLock)
-            {
-                this.monitoredSessions[sessionId].pollImmediatly = true;
-            }
-            lock (this.pollingLock)
-            {
-                Monitor.Pulse(pollingLock);
-            }
-        }
-
         /// <summary>
-        /// The core queue processing method
+        /// The core queue processing method, cycles through monitored sessions and creates a stream for them if not already.
```

Comment:
One thing to think about, why do we even need to do this? Why don't we start the session right away when we add it to the dictionary? It's fine to leave it with the existing behavior for now, but a good follow-up task would be to see if this is actually necessary (and if it is then make sure it's documented in the code why we do this)

Reply:
Likely it is not necessary, but I will investigate in another follow up task.